libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -31,23 +31,35 @@ using namespace solidity;
 using namespace solidity::util;
 using namespace solidity::frontend;
 
-string IRGenerationContext::addLocalVariable(VariableDeclaration const& _varDecl)
+IRVariable const& IRGenerationContext::addFunctionParameter(VariableDeclaration const& _varDecl)
 {
-	solUnimplementedAssert(
-		_varDecl.annotation().type->sizeOnStack() == 1,
-		"Multi-slot types not yet implemented."
+	auto const& [it, didInsert] = m_localVariables.emplace(
```

Comment:
What is this? Can we do this less-performing but more readable? :)

Reply:
Which part, the return value or the arguments :-)?
Here specifically it's actually not easy doing it differently, since the variables are (intentionally) not copiable and not default-constructible...